### PR TITLE
V1.3 remark commands

### PR DESCRIPTION
**Pull request by inezkok**
Status: COMPLETED

Tasks:
- [x] Add Remark class
- [x] Add RemarkList class
- [x] Implement AddRemarkCommand & AddRemarkCommandParser
- [x] Implement DeleteRemarkCommand & DeleteRemarkCommandParser
- [x] AddRemarkCommandTest & AddRemarkCommandParserTest
- [x] DeleteRemarkCommandTest & DeleteRemarkCommandParserTest

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddRemarkCommand.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            + PREFIX_REMARK + "Good worker";
+
+    public static final String MESSAGE_SUCCESS = "New remark added: %1$s";
+    public static final String MISSING_REMARK = "No remark given!";
+    public static final String MESSAGE_DUPLICATE_REMARK = "This remark already exists";
+    private final Id targetId;
+    private final Remark remark;
+
+    /**
+     * Creates an AddRemarkCommand to add the specified {@code Employee}
```
Comment (miljyy) [inline src/main/java/seedu/address/logic/commands/AddRemarkCommand.java:41]: minor nitpick, but I believe it should be "Creates an AddRemarkCommand to add to the specified {@code Employee}"

File: src/main/java/seedu/address/logic/commands/AddRemarkCommand.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        }
+        throw new CommandException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_USAGE));
+    }
+
+    /**
+     * Adds remarks to the RemarkList.
+     * @param currentList The RemarkList to add to.
+     * @param remark The Remark to be added.
+     * @return RemarkList with added Remark.
+     * @throws CommandException If the user input does not conform the expected format
```
Comment (miljyy) [inline src/main/java/seedu/address/logic/commands/AddRemarkCommand.java:76]: Perhaps can modify the Javadoc to be "throws CommandException If the remark already exists/has a duplicate"?

File: src/main/java/seedu/address/logic/commands/DeleteRemarkCommand.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    /**
+     * Deletes remark from the RemarkList.
+     * @param currentList The RemarkList to be deleted from.
+     * @param remark The Remark to be deleted.
+     * @return RemarkList without deleted Remark.
+     * @throws CommandException If the user input does not conform the expected format
+     */
+    private RemarkList deleteRemarkToList(RemarkList currentList, Remark remark) throws CommandException {
```
Comment (miljyy) [inline src/main/java/seedu/address/logic/commands/DeleteRemarkCommand.java:78]:
```suggestion
    private RemarkList deleteRemarkFromList(RemarkList currentList, Remark remark) throws CommandException {
```

File: src/main/java/seedu/address/logic/parser/AddRemarkCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import seedu.address.model.remark.Remark;
+
+/**
+ * Parses input arguments and creates a new AddRemarkCommand object
+ */
+public class AddRemarkCommandParser implements Parser<AddRemarkCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddLeaveCommand
+     * and returns a AddLeaveCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
```
Comment (miljyy) [inline src/main/java/seedu/address/logic/parser/AddRemarkCommandParser.java:22]:
```suggestion
     * Parses the given {@code String} of arguments in the context of the AddRemarkCommand
     * and returns a AddRemarkCommand object for execution.
     * @throws ParseException if the user input does not conform to the expected format
```

File: src/main/java/seedu/address/logic/parser/AddRemarkCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddRemarkCommand.MESSAGE_USAGE), pe);
+        }
+
+        if (argMultimap.getValue(PREFIX_REMARK).isEmpty()) {
+            throw new ParseException(AddRemarkCommand.MISSING_REMARK);
+        }
+
+        if (argMultimap.getValue(PREFIX_REMARK).isPresent()) {
```
Comment (miljyy) [inline src/main/java/seedu/address/logic/parser/AddRemarkCommandParser.java:47]: Is there still a need for this check if there is already the `argMultimap.getValue(PREFIX_REMARK).isEmpty()` check?

File: src/main/java/seedu/address/logic/parser/DeleteRemarkCommandParser.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import seedu.address.model.remark.Remark;
+
+/**
+ * Parses input arguments and creates a new DeleteRemarkCommand object
+ */
+public class DeleteRemarkCommandParser implements Parser<DeleteRemarkCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the DeleteLeaveCommand
+     * and returns a DeleteLeaveCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
```
Comment (miljyy) [inline src/main/java/seedu/address/logic/parser/DeleteRemarkCommandParser.java:22]:
```suggestion
     * Parses the given {@code String} of arguments in the context of the DeleteRemarkCommand
     * and returns a DeleteRemarkCommand object for execution.
     * @throws ParseException if the user input does not conform to the expected format
```

File: src/main/java/seedu/address/logic/parser/DeleteRemarkCommandParser.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteRemarkCommand.MESSAGE_USAGE), pe);
+        }
+
+        if (argMultimap.getValue(PREFIX_REMARK).isEmpty()) {
+            throw new ParseException(DeleteRemarkCommand.MISSING_REMARK);
+        }
+
+        if (argMultimap.getValue(PREFIX_REMARK).isPresent()) {
```
Comment (miljyy) [inline src/main/java/seedu/address/logic/parser/DeleteRemarkCommandParser.java:47]: Similar to the comment in AddRemarkCommandParser

File: src/main/java/seedu/address/model/department/Department.java [hunk 8]
```diff
@@ -4,29 +4,29 @@
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
 /**
- * Represents a Tag in the address book.
+ * Represents a Department in the address book.
```
Comment (miljyy) [inline src/main/java/seedu/address/model/department/Department.java:7]: Great work modifying from Tag to Department!

File: src/main/java/seedu/address/model/remark/Remark.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+     * Returns true if remark is non-empty.
+     *
+     * @param test Remark to be tested.
+     * @return True if the remark is non-empty.
+     */
+    public static boolean isValidRemark(String test) {
+        if (!test.isBlank()) {
+            return true;
+        } else {
+            return false;
```
Comment (miljyy) [inline src/main/java/seedu/address/model/remark/Remark.java:62]:
```suggestion
        return !test.isBlank();
```

**Review by miljyy — COMMENTED** [review]
Overall, well done adding both the add and delete remark commands, as well as integrating the remark attribute into the employee!

File: src/main/java/seedu/address/model/department/Department.java [hunk 10]
```diff
@@ -4,29 +4,29 @@
...
-        this.departmentName = tagName;
+    public Department(String departmentName) {
+        requireNonNull(departmentName);
+        checkArgument(isValidDepartmentName(departmentName), MESSAGE_CONSTRAINTS);
+        this.departmentName = departmentName;
     }
 
     /**
-     * Returns true if a given string is a valid tag name.
+     * Returns true if a given string is a valid department name.
```
Comment (s0ngyang) [inline src/main/java/seedu/address/model/department/Department.java:29]: Good catch, thanks for updating the variable name

**Review by s0ngyang — APPROVED** [review]
Test cases are comprehensive. LGTM!

**Pull request merged**